packages/now-routing-utils/src/index.ts

```diff
@@ -135,38 +155,88 @@ export function getTransformedRoutes({
         message: 'Cannot define both `routes` and `rewrites`',
       });
     }
-  } else {
-    routes = [];
-    if (typeof cleanUrls !== 'undefined') {
-      routes.push(...convertCleanUrls(cleanUrls));
+    if (errors.length > 0) {
+      const error = createNowError(
+        'invalid_keys',
+        'Cannot mix legacy routes with new keys',
+        errors
+      );
+      return { routes, error };
     }
-    if (typeof trailingSlash !== 'undefined') {
-      routes.push(...convertTrailingSlash(trailingSlash));
+    return normalizeRoutes(routes);
+  }
+
+  routes = [];
+
+  if (typeof cleanUrls !== 'undefined') {
+    const normalized = normalizeRoutes(convertCleanUrls(cleanUrls));
+    if (normalized.error) {
+      normalized.error.code = 'invalid_clean_urls';
+      return { routes, error: normalized.error };
     }
-    if (typeof redirects !== 'undefined') {
-      routes.push(...convertRedirects(redirects));
+    routes.push(...(normalized.routes || []));
+  }
+
+  if (typeof trailingSlash !== 'undefined') {
+    const normalized = normalizeRoutes(convertTrailingSlash(trailingSlash));
+    if (normalized.error) {
+      normalized.error.code = 'invalid_trailing_slash';
+      return { routes, error: normalized.error };
     }
-    if (typeof headers !== 'undefined') {
-      routes.push(...convertHeaders(headers));
+    routes.push(...(normalized.routes || []));
+  }
+
+  if (typeof redirects !== 'undefined') {
+    const code = 'invalid_redirects';
+    const errors = redirects
+      .map(r => checkRegexSyntax(r.source))
+      .filter(notEmpty);
+    const error = createNowError(
+      code,
+      'Redirect `source` contains invalid regex',
+      errors
+    );
+    if (error) {
```

Comment:
I find this a bit confusing at first given `error` doesn't really seem like it'd be a conditional

Reply:
@dav-is I'm open to suggestions. I tried a couple different things and I didn't like any of them.

I think the really strange thing about this API is that there is a top-level error with nested errors underneath. It would be great to remove that and just return one error.

Reply:
It seems like to me that it was designed like that so you can show all the errors related to your routes without fixing one, redeploying, fixing another, redeploying and so on

Reply:
Why not:
```
if (errors.length > 0) {
  return { 
    routes,
    error: createNowError(
      'invalid_redirects',
      'Redirect `source` contains invalid regex',
      errors
    )
  }
}
```

Reply:
Your example seem to be just as verbose as the original but I will change it 😄

Reply:
What about:
```
if (errors.length > 0) {
  return createNowError(
    'invalid_redirects',
    'Redirect `source` contains invalid regex',
    routes,
    errors
  )
}
```
All I'm after is the conditional being readable `if (errors.length > 0) {`

`const error = createNewError()` makes me think `error` will always be an error